packages/eas-cli/src/commands/build/index.ts

```diff
@@ -135,6 +142,15 @@ export default class Build extends EasCommand {
       );
       Log.newLine();
     }
+    if (flags['resource-class']) {
+      switch (flags['resource-class']) {
+        case UserInputResourceClass.DEFAULT:
+        case UserInputResourceClass.LARGE:
+          break;
+        default:
+          Errors.error(`Invalid resource-class: '${flags['resource-class']}'`, { exit: 1 });
+      }
+    }
```

Comment:
```suggestion
    if (flags['resource-class'] && !Object.values(UserInputResourceClass).includes(flags['resource-class'])) {
      Errors.error(`Invalid resource-class: '${flags['resource-class']}'`, { exit: 1 });
    }
```

The code you wrote is really tricky and hard to understand. Also, if you wanted to add a new value to the enum you would need to know about this code.